### PR TITLE
Hugenum math in Storm (SYN-3350)

### PR DESCRIPTION
**Pull request by Cisphyx**

### CODE REVIEW

File: synapse/lib/stormtypes.py [hunk 1]
```diff
@@ -8489,6 +8636,38 @@ def fromprim(valu, path=None, basetypes=True):
...
+async def tocmprvalu(valu):
+
+    if isinstance(valu, (str, int, bool, float, bytes, types.AsyncGeneratorType, types.GeneratorType, Number)) or valu is None:
+        return valu
+
+    if isinstance(valu, (tuple, list)):
+        retn = []
+        for v in valu:
+            try:
+                retn.append(await toprim(v))
```
Comment (invisig0th) [inline synapse/lib/stormtypes.py:8648]: Do these cases need to be tocmprvalu() recursion like they are in the toprim case?

File: synapse/tests/test_lib_stormtypes.py [hunk 2]
```diff
@@ -5186,12 +5201,23 @@ async def test_stormtypes_tofoo(self):
...
         self.true(await s_stormtypes.tobool(s_stormtypes.Str('asdf')))
         self.false(await s_stormtypes.tobool(s_stormtypes.Str('')))
 
+        self.eq(20, await s_stormtypes.tonumber('20'))
+        self.eq(20.1, await s_stormtypes.tonumber(20.1))
+        self.eq(20.1, await s_stormtypes.tonumber('20.1'))
+        self.eq(20.1, await s_stormtypes.tonumber(s_stormtypes.Number('20.1')))
+
+        self.eq(True, await s_stormtypes.tocmprvalu(boolprim))
+        self.eq((1, 3), await s_stormtypes.tocmprvalu([1, s_exc.SynErr, 3]))
```
Comment (invisig0th) [inline synapse/tests/test_lib_stormtypes.py:5218]: These test cases give me pause. Lets discuss :)

**Review by invisig0th — CHANGES_REQUESTED** [review]
Really digging this. Couple discussion points.

File: synapse/tests/test_lib_stormtypes.py (hunk 2 above)
Comment (Cisphyx) [reply synapse/tests/test_lib_stormtypes.py]: updated to allow these through

**Review by invisig0th — APPROVED** [review]

**Pull request merged**